File: Can_Place_Flowers.go
package main

import "fmt"

func main() {
	// flowerbed := []int{1, 0, 0, 0, 1}
	// n := 1

	flowerbed := []int{0, 0, 1, 0, 1}
	n := 1
	fmt.Println(canPlaceFlowers(flowerbed, n))
}

func canPlaceFlowers(flowerbed []int, n int) bool {
	i := 1
	mk := make([]int, len(flowerbed)+2)

	for j := 1; j < len(flowerbed)+1; j++ {
		mk[j] = flowerbed[j-1]
	}

	for i < len(mk)-1 {
		if mk[i-1] == 0 && mk[i] == 0 && mk[i+1] == 0 {
			mk[i] = 1
			n -= 1

		}
		if n <= 0 {
			return true
		}
		i += 1
	}

	return false

}

File: Can_Place_Flowers/Can_Place_Flowers.py
# flowerbed = [1,0,0,0,1]
# n = 1
# flowerbed = [1,0,0,0,1]
# n = 2

flowerbed = [1,0,0,0,0,1]
n=2

def canPlaceFlowers(flowerbed,n):
    flowerbed = [0] + flowerbed + [0]
    # flowerbed.append(0)
    print(flowerbed)
    totalL = len(flowerbed) - 1
    i = 1

    while i <= totalL -1:
        if flowerbed[i-1] == 0 and flowerbed[i] ==0 and flowerbed[i+1] == 0:
            flowerbed[i] = 1
            n -=1
        
        if n <= 0:
            return True

        i+=1
    
    print(flowerbed)
    return False
print(canPlaceFlowers(flowerbed,n))